assets/src/design-system/components/search/test/search.js

```diff
@@ -0,0 +1,272 @@
+    });
+
+    expect(input).toBeInTheDocument();
+  });
+
+  it('should show <Search /> menu when input has 1 or more characters', () => {
+    const { getByRole } = renderWithProviders(
+      <Search
+        emptyText="No options available"
+        options={basicDropDownOptions}
+        ariaInputLabel="label"
+      />
+    );
+
+    const input = getByRole('combobox');
+    expect(input).toBeInTheDocument();
+    act(() => {
+      fireEvent.click(input);
+      fireEvent.change(input, { target: { value: 'bruce' } });
+
+      // wait for debounced callback to allow a select click handler to process
+      jest.runOnlyPendingTimers();
+    });
+
+    const menu = getByRole('listbox');
+    expect(menu).toBeInTheDocument();
+  });
+
+  it('should show an active icon on list item that is active', () => {
+    const { getByRole } = renderWithProviders(
+      <Search
+        emptyText="No options available"
+        ariaInputLabel="label"
+        isKeepMenuOpenOnSelection={false}
+        options={basicDropDownOptions}
+        selectedValue={basicDropDownOptions[2]}
+      />
+    );
+
+    const input = getByRole('combobox');
+    expect(input).toBeInTheDocument();
+
+    act(() => {
+      fireEvent.click(input);
+
+      // wait for debounced callback to allow a select click handler to process
+      jest.runOnlyPendingTimers();
+    });
+
+    const activeMenuItem = getByRole('option', {
+      name: `Selected ${basicDropDownOptions[2].label}`,
+    });
+    expect(activeMenuItem).toBeInTheDocument();
+  });
+
+  // Mouse events
+  it('should not expand menu when disabled is true', () => {
+    const { getByRole, queryAllByRole } = renderWithProviders(
+      <Search
+        options={basicDropDownOptions}
+        ariaInputLabel="my label"
+        disabled
+      />
+    );
+
+    const input = getByRole('combobox');
+    expect(input).toBeInTheDocument();
+
+    act(() => {
+      fireEvent.click(input);
+
+      // wait for debounced callback to allow a select click handler to process
+      jest.runOnlyPendingTimers();
+    });
+
+    const menu = queryAllByRole('listbox');
+    expect(menu).toStrictEqual([]);
+  });
+
+  it('should not expand menu with selected value when disabled is true', () => {
+    const { getByRole, queryAllByRole } = renderWithProviders(
+      <Search
+        options={basicDropDownOptions}
+        selectedValue={basicDropDownOptions[2]}
+        ariaInputLabel="my label"
+        disabled
+      />
+    );
+
+    const input = getByRole('combobox');
+    expect(input).toBeInTheDocument();
+
+    act(() => {
+      fireEvent.click(input);
+
+      // wait for debounced callback to allow a select click handler to process
+      jest.runOnlyPendingTimers();
+    });
+
+    const menu = queryAllByRole('listbox');
+    expect(menu).toStrictEqual([]);
+  });
+
+  it('should trigger onMenuItemClick from input when input has value', () => {
```

Comment:
💭 I think the title of this test is a little confusing to me. If you like this suggestion, maybe we could update it to:
```suggestion
  it('should trigger onMenuItemClick from input the user enters a value that exists in the dropdown and presses the `Enter` key', () => {
```

Reply:
maybe that's too verbose 🤔. I think it's important to convey that the value the user entered exists and they are hitting enter while in the input, rather than selecting it in the dropdown.

Reply:
Good call. I updated it to: `should trigger onMenuItemClick when "enter" is hit and input has value` and added a test above it showing that `onMenuItemClick` is not triggered when the input is empty.